tests/Util/CliOutputHelperTest.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Tests\Util;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\MakerBundle\Util\CliOutputHelper;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ */
+class CliOutputHelperTest extends TestCase
+{
+    /**
+     * @dataProvider cliEnvProvider
+     */
+    public function testCorrectPrefixReturnedBasedOnInputMethod(string $cliVersion, string $cliBinary, string $expectedPrefix): void
+    {
+        putenv(sprintf('%s=SYMFONY', $cliBinary));
+        putenv(sprintf('%s=1.0.0', $cliVersion));
+
+        self::assertSame($expectedPrefix, CliOutputHelper::getCommandPrefix());
+
+        putenv($cliVersion);
+        putenv($cliBinary);
+    }
+
+    public function cliEnvProvider(): \Generator
+    {
+        yield 'Using Symfony CLI' => ['SYMFONY_CLI_VERSION', 'SYMFONY_CLI_BINARY_NAME', 'symfony console'];
+        yield 'Without Symfony CLI' => ['ARBITRARY_VERSION', 'ARBITRARY_BIN_NAME', 'php bin/console'];
```

Comment:
We're basically testing how things look when you set the env vars vs do NOT set the env vars. Setting some env vars in both cases (but using made-up names in the 2nd case, as a way to avoid setting the env var in question) smells a touch funny :). This might be more clear without a data provider - just with 2 test methods, where one doesn't set any env vars.

Reply:
good catch